sw/device/tests/sram_ctrl_smoketest.c

```diff
@@ -0,0 +1,75 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+// Write / Read small chunks of data to SRAM to test
+// basic functionality of sram_ctrl
+static void write_read_check(mmio_region_t base_addr_region,
+                             const char *sram_name) {
+  uint32_t rw_data_32;
+  for (int i = 0; i < 16; i++) {
+    rw_data_32 = (i << 26 | i << 19 | i << 11 | i << 3 | i >> 5);
```

Comment:
Can this be factored out into a helper function that specifies the significance of the sequence?

Note also that this may potentially result in shifting into the sign bit, which is undefined behavior; moreover, note that
`i >> 5` is always zero if `i < 16`.

Reply:
Hi Miguel, thanks for your comments. There is no significance to this sequence. Please suggest a function to fill up the data or I could use an array of constants. What would you do? I will make the changes for the other comments you have made as they all make sense and I will try to remember to take these suggestions into the other code I write.